contents/blog/the-posthog-array-1-41-0.md

```diff
@@ -0,0 +1,146 @@
+---
+date: 2022-10-31
+title: "Array 1.41.0: Enable hedgehog mode"
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+categories: ['Release notes', 'Product updates']
+featuredImage: ../images/blog/posthog-array-blog.png
+featuredImageType: full
+---
+
+Want to know more about what we're up to? [Subscribe to HogMail, our newsletter](/newsletter), which we send every two weeks!
+
+> Running a self-hosted instance? Check out our [guide to upgrading PostHog](/docs/runbook/upgrading-posthog).
+
+> **IMPORTANT NOTE FOR SELF-HOSTED USERS:** Self-hosted users should check `alias` usage before [upgrading PostHog](/docs/runbook/upgrading-posthog) due to changes involving persons on events. For example, assuming `email` is used as the identified user id, then: 
+
+```
+identify(email)                                   # in the frontend
+alias(backend_unique_id, email)      # in the backend - THIS WILL NOT WORK
+alias(email, backend_unique_id2)    # in the backend - the right order
+```
+After updating to 1.41.0, check [ingestion warnings](#new-ingestion-warnings) and run async migrations 0005, 0006 and 0007.
```

Comment:
This is also a bit more complex. Essentially the guide is:

First check ingestion warnings & fix everything there.
Note that if you're using alias in the wrong order your users will not be merged anymore from those bad calls, so if fixing this will take a long time you might want to downgrade to 1.40 - not sure if want to call this out or not 🤷‍♀️ 

Then
# if you haven't ran 0007 (and potentially not 0005, 0006 either) async migration:
Run the async migrations if you haven't ran them yet, they are required to be completed before upgrading to the next version, you can do so on the async migrations UI at `<your-posthog-site>/instance/async_migrations`.

Completing async migration 0007 enables persons-on-events. 
You can disable it from instance settings `PERSON_ON_EVENTS_ENABLED` if your queries aren't working properly - this can be mentioned in the section we have for persons-on-events.

# if async migration 0007 is completed 

IMPORTANT: If you either changed anything about alias or based of ingestion warnings or are using group you will need to re-run 0007 (make sure to do this after fixing all the problems from ingestion warnings).

How to re-run 0007:
1. connect to postgres pod: https://posthog.com/docs/self-host/deploy/troubleshooting#how-do-i-connect-to-postgres
2. update the 0007 async migration to be Not Started (it's safe it was in Completed state before), by running:
```
UPDATE posthog_asyncmigration SET status = 0 WHERE name = '0007_persons_and_groups_on_events_backfill' AND status = 2;
```
3. click on the run button on the async migrations UI at `<your-posthog-site>/instance/async_migrations`

Reply:
Thanks Tiina - I've made some updates.